### PR TITLE
[proxima-direct-io-kafka] #134 throw IllegalArgumentException on non-existing topic

### PR DESCRIPTION
**Pull request by je-ik**

### CODE REVIEW

**Comment by je-ik** [comment]
Closes O2-Czech-Republic#134

File: direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java [hunk 1]
```diff
@@ -636,6 +637,16 @@ public void waitUntilReady() throws InterruptedException {
     return consumer;
   }
 
+  @VisibleForTesting
+  void validateTopic(KafkaConsumer<?, ?> consumer, String topicToValidate) {
+    List<PartitionInfo> partitions = consumer.partitionsFor(topicToValidate);
```
Comment (dmvk) [inline direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java]: nit: this is only necessary when auto create of topic is disabled

File: direct/io-kafka/src/test/java/cz/o2/proxima/direct/kafka/LocalKafkaCommitLogDescriptorTest.java [hunk 2]
```diff
@@ -1632,6 +1633,25 @@ public boolean onError(Throwable error) {
...
   }
 
+  @Test
+  public void testObserveOnNonExistingTopic() throws InterruptedException {
+    Accessor accessor = kafka.createAccessor(direct, entity, storageUri, partitionsCfg(3));
+    LocalKafkaLogReader reader = accessor.newReader(context());
+    try {
+      // need this to initialize the consumer
+      assertNotNull(reader.getPartitions());
+      reader.validateTopic(reader.getConsumer(), "non-existing-topic");
```
Comment (dmvk) [inline direct/io-kafka/src/test/java/cz/o2/proxima/direct/kafka/LocalKafkaCommitLogDescriptorTest.java]: how does this test that the validateTopic method gets invoked at the right moment?

Comment (je-ik) [reply direct/io-kafka/src/test/java/cz/o2/proxima/direct/kafka/LocalKafkaCommitLogDescriptorTest.java]: It does not. The problem is that there is no way to inject "wrong" topic to the test (mock) reader.

**Pull request merged**